src/work/test/WorkTests.cpp

```diff
@@ -315,7 +317,8 @@ TEST_CASE("work with children", "[work]")
 
     SECTION("success")
     {
-        while (!wm.allChildrenSuccessful())
+        while (!wm.allChildrenSuccessful() ||
+               wm.getState() == TestBasicWork::State::WORK_RUNNING)
```

Comment:
This is needed: the old code was assuming that wm would go go through multiple transitions in one call to `crank` (and it was because `crank` would cause up to 100 transitions).

The way `Work` works, is that it calls `doWork` for the `Work` instance itself in a different transition than for its children.